### DISCUSSION TITLE
Array of enum validation

### DISCUSSION THREAD

**Discussion by AmirRezaM75** [spatie/laravel-data#564] [Q&A]
Is there any predefined attribute that checks if the given input is an array of specific enum?
If NO. Do you have any plan for that? Is this feature valid for a pull request or do you prefer to not include it in package's source code and handle it with custom casts?

```php
class PlaceOrdeRequest extends Data
{
    /** @param ShipmentMethod[] $methods */
    public function __construct(
        #[Required]
        public readonly array $methods,
    ) {
    }
}
```

**Expectation**:
`$request->methods` returns array of ShipmentMethod enums

**Comment by rubenvanassche** [comment]
At the moment such a thing does not exist, you could always add such an attribute in your own project. Not sure if we want this in the package but it could be useful. My first idea would be an `ArrayOf` attribute which can have several rules:

```php
#[ArrayOf(new Enum(ShipmentMethod::class))] // ['methods.*' => 'enum: ShipmentMethod']
```

```
#[ArrayOf(new IntegerType, new Min(10), new Max(20))]  // ['methods.*' => ['integer', 'min:10', 'max:20']]
```

Not sure about the name and how to implement it but looks like an interesting experiment.

**Comment by raimondsL** [comment]
```
request()->validate([
   'unit.*' => [Rule::enum(UnitEnum::class)],
]);
```

> {"message":"The selected unit.2 is invalid.","errors":{"unit.2":["The selected unit.2 is invalid."]}}

**Comment by hfoletto** [comment]
Hello there,

Is there an update on this subject?

Thank you!
